src/main/java/org/eclipse/xpanse/terraform/boot/api/TerraformApiController.java

```diff
@@ -107,41 +111,77 @@ public TerraformResult destroy(
             @Parameter(name = "module_directory",
                     description = "directory name where the Terraform module files exist.")
             @PathVariable("module_directory") String moduleDirectory,
-            @Valid @RequestBody TerraformDestroyRequest terraformDestroyRequest) {
-        return this.terraformExecutor.destroy(terraformDestroyRequest, moduleDirectory);
+            @Valid @RequestBody
+                    TerraformDestroyFromDirectoryRequest terraformDestroyFromDirectoryRequest) {
+        return this.terraformExecutor.destroyFromDirectory(terraformDestroyFromDirectoryRequest,
+                moduleDirectory);
     }
 
     /**
-     * Method to async deploy resources requested in a workspace.
+     * Method to deploy resources by scripts.
      *
+     * @return Returns the status of the deployment.
      */
     @Tag(name = "Terraform", description = "APIs for running Terraform commands")
-    @Operation(description = "async deploy resources via Terraform")
-    @PostMapping(value = "/deploy/async/{module_directory}", produces =
+    @Operation(description = "Deploy resources via Terraform")
+    @PostMapping(value = "/deploy/scripts/{module_directory}", produces =
             MediaType.APPLICATION_JSON_VALUE)
-    @ResponseStatus(HttpStatus.ACCEPTED)
-    public void asyncDeploy(
+    @ResponseStatus(HttpStatus.OK)
+    public TerraformResult deployWithScripts(
             @Parameter(name = "module_directory",
                     description = "directory name where the Terraform module files exist.")
             @PathVariable("module_directory") String moduleDirectory,
-            @Valid @RequestBody TerraformAsyncDeployRequest asyncDeployRequest) {
-        terraformExecutor.asyncDeploy(asyncDeployRequest, moduleDirectory);
+            @Valid @RequestBody TerraformDeployWithScriptsRequest request) {
+        return terraformExecutor.deployWithScripts(request, moduleDirectory);
     }
 
     /**
-     * Method to async destroy resources requested in a workspace.
+     * Method to destroy resources by scripts.
      *
+     * @return Returns the status of the destroy.
+     */
+    @Tag(name = "Terraform", description = "APIs for running Terraform commands")
+    @Operation(description = "Deploy resources via Terraform")
+    @PostMapping(value = "/destroy/scripts/{module_directory}", produces =
```

Comment:
in case of scripts, we do not need the  "module_directory" parameter. Applies to all "script" methods.